app/Http/Controllers/Three/SignupsController.php

```diff
@@ -38,4 +38,22 @@ public function index(Request $request)
 
         return $this->paginatedCollection($query, $request);
     }
+
+    /**
+     * Returns a specific signup(s).
+     * GET /signups/:id
+     *
+     * @param Request $request
+     * @return ]Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $signup = Signup::find($id);
+
+        if ($signup) {
+            return $this->item($signup);
+        } else {
+            throw (new ModelNotFoundException)->setModel('Signup');
```

Comment:
We're not importing this class at the top of this file, so that's why you're seeing a 500 error with "Class not found" on 404s, instead of an actual 404 response code.

Let's switch to using `findOrFail` on the model to clean things up a bit & get the proper exception:

```php
$signup = Signup::findOrFail($id);

return $this->item($signup);
```